src/Microsoft.AspNetCore.OData/EnableQueryAttribute.cs

```diff
@@ -234,6 +235,26 @@ public override void OnActionExecuted(ActionExecutedContext actionExecutedContex
                 // been applied to the result; make sure it's also successful.
                 ObjectResult responseContent = actionExecutedContext.Result as ObjectResult;
 
+                if (responseContent == null)
+                {
+                    Type actionResultType = actionExecutedContext.Result.GetType();
+                    Type[] actionResultGenericArgs = actionResultType.GetGenericArguments();
+                    Type createdODataResultType = actionResultGenericArgs.Length > 0 ?
+                        typeof(CreatedODataResult<>).MakeGenericType(actionResultGenericArgs[0]) : null;
```

Comment:
can we have more than one `actionResultGenericArgs` here? Why do we choose to get the one at position `zero`?

Reply:
[CreatedODataResult](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnet.odata.results.createdodataresult-1) has only one argument